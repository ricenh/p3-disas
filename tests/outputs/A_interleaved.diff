--- outputs/A_interleaved.txt	2023-11-06 15:39:34.710126763 -0500
+++ expected/A_interleaved.txt	2023-10-10 11:54:35.000000000 -0400
@@ -11,6 +11,12 @@
   0x200:                               | .pos 0x200 data
   0x200: 10 00 00 00 00 00 00 00       |   .quad 0x10
 
+  0x300:                               | .pos 0x300 rodata
+  0x300: 74 65 73 74 00                |   .string "test"
+
   0x500:                               | .pos 0x500 data
   0x500: 78 56 34 12 00 00 00 00       |   .quad 0x12345678
 
+  0x600:                               | .pos 0x600 rodata
+  0x600: 61 67 61 69 6e 00             |   .string "again"
+
