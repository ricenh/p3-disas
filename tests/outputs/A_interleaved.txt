Usage: ../y86 <option(s)> mini-elf-file
 Options are:
  -h      Display usage
  -H      Show the Mini-ELF header
  -a      Show all with brief memory
  -f      Show all with full memory
  -s      Show the program headers
  -m      Show the memory contents (brief)
  -M      Show the memory contents (full)
  -d      Disassemble code contents
  -D      Disassemble data contents
