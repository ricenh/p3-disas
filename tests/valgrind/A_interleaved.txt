==1081034== Memcheck, a memory error detector
==1081034== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1081034== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1081034== Command: ../y86 -d -D inputs/interleaved.o
==1081034== 
Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 00                            |   halt

  0x400:                               | .pos 0x400 code
  0x400: 10                            |   nop
  0x401: 00                            |   halt

Disassembly of data contents:
  0x200:                               | .pos 0x200 data
  0x200: 10 00 00 00 00 00 00 00       |   .quad 0x10

  0x500:                               | .pos 0x500 data
  0x500: 78 56 34 12 00 00 00 00       |   .quad 0x12345678

==1081034== 
==1081034== HEAP SUMMARY:
==1081034==     in use at exit: 0 bytes in 0 blocks
==1081034==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1081034== 
==1081034== All heap blocks were freed -- no leaks are possible
==1081034== 
==1081034== For lists of detected and suppressed errors, rerun with: -s
==1081034== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
