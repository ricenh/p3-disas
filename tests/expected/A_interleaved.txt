Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 00                            |   halt

  0x400:                               | .pos 0x400 code
  0x400: 10                            |   nop
  0x401: 00                            |   halt

Disassembly of data contents:
  0x200:                               | .pos 0x200 data
  0x200: 10 00 00 00 00 00 00 00       |   .quad 0x10

  0x300:                               | .pos 0x300 rodata
  0x300: 74 65 73 74 00                |   .string "test"

  0x500:                               | .pos 0x500 data
  0x500: 78 56 34 12 00 00 00 00       |   .quad 0x12345678

  0x600:                               | .pos 0x600 rodata
  0x600: 61 67 61 69 6e 00             |   .string "again"

